experimental/dds/tree2/src/feature-libraries/editable-tree/editableField.ts

```diff
@@ -47,13 +48,34 @@ import {
 import { makeTree } from "./editableTree";
 import { ProxyTarget } from "./ProxyTarget";
 
+/* eslint-disable no-param-reassign */
+/* eslint-disable no-bitwise */
+
+/**
+ * Coerces the given 'index' to an Int32.  This includes the following behaviors:
+ *
+ * - +/-Infinity is preserved
+ * - \{ undefined, null, false, NaN \} map to 0
+ * - \{ true \} maps to 1
+ * - Strings containing numeric digits are parsed (e.g. "-0.1e2" -\> -10)
+ * - Non-parsable strings are coerced to 0
```

Comment:
you carefully define a bunch of cases which can never be used because of the type of index, but don't define what thing allowed by the type system do, like 0.5, integers larger than (or equal to) the length.

Reply:
Interesting point.  I don't know how much pre-existing code we would break if we do not all of the Array behavior.